### PR TITLE
Fix compiler warning and potential string buffer overflow

### PR DESCRIPTION
**Pull request by soligen2010**
### Description

Looks like version 2.1.2.2 is coming soon, and since the latest bugfix has these compiler warning, I thought you may want to include these.  They are simple changes.

The lastEncoderDiff warning is just a warning message that I don't think would affect anything.  It just  conditionally removes un-used code

The Status Screen change is a buffer overflow warning. The buffer was 10, but the compiler warning said it could be up to 13.

### Requirements

I don't know what all boards may be affected.  I use SKR 1.4 Turbo

### Benefits

1) Removes compiler warning that may confuse some users
2) Prevents a potential buffer overflow that may cause a display or stability issue. 

### Configurations

I tried to upload my config files, but GitHub says it is not a supported file type.  Hopefully this is simple enough that you don't need them, but let me know if you do.

### Related Issues

None I know of.

### CODE REVIEW

File: Marlin/src/lcd/marlinui.cpp [hunk 1]
```diff
@@ -1035,9 +1035,9 @@ void MarlinUI::init() {
         uint8_t abs_diff = ABS(encoderDiff);
 
         #if ENCODER_PULSES_PER_STEP > 1
-          static int8_t lastEncoderDiff;
+          TERN_(HAS_TOUCH_SLEEP, static int8_t lastEncoderDiff);
           TERN_(HAS_TOUCH_SLEEP, if (lastEncoderDiff != encoderDiff) wakeup_screen());
-          lastEncoderDiff = encoderDiff;
+          TERN_(HAS_TOUCH_SLEEP, lastEncoderDiff = encoderDiff);
         #endif
```
Comment (classicrocker883) [inline Marlin/src/lcd/marlinui.cpp]: how about changing it to this?

```diff
         #if ENCODER_PULSES_PER_STEP > 1
-          TERN_(HAS_TOUCH_SLEEP, static int8_t lastEncoderDiff);
-          TERN_(HAS_TOUCH_SLEEP, if (lastEncoderDiff != encoderDiff) wakeup_screen());
-          TERN_(HAS_TOUCH_SLEEP, lastEncoderDiff = encoderDiff);
         #endif

         #if ENCODER_PULSES_PER_STEP > 1
+          #if HAS_TOUCH_SLEEP
+            static int8_t lastEncoderDiff);
+            if (lastEncoderDiff != encoderDiff) wakeup_screen());
+            lastEncoderDiff = encoderDiff);
+          #endif
         #endif

**Review by classicrocker883 — CHANGES_REQUESTED** [review]

File: Marlin/src/lcd/marlinui.cpp (hunk 1 above)
Comment (soligen2010) [reply Marlin/src/lcd/marlinui.cpp]: Sounds good.  The change is made.

File: Marlin/src/lcd/marlinui.cpp [hunk 2]
```diff
@@ -1035,11 +1035,13 @@ void MarlinUI::init() {
         uint8_t abs_diff = ABS(encoderDiff);
 
         #if ENCODER_PULSES_PER_STEP > 1
-          static int8_t lastEncoderDiff;
-          TERN_(HAS_TOUCH_SLEEP, if (lastEncoderDiff != encoderDiff) wakeup_screen());
-          lastEncoderDiff = encoderDiff;
+          #if HAS_TOUCH_SLEEP
+            static int8_t lastEncoderDiff);
+            if (lastEncoderDiff != encoderDiff) wakeup_screen());
+            lastEncoderDiff = encoderDiff);
```
Comment (XDA-Bam) [inline Marlin/src/lcd/marlinui.cpp]: Do I miss something or are there closing brackets that don't belong there at the end of each line?

Comment (classicrocker883) [reply Marlin/src/lcd/marlinui.cpp]: you're correct. looks like they were forgotten.

File: Marlin/src/lcd/marlinui.cpp (hunk 1 above)
Comment (classicrocker883) [reply Marlin/src/lcd/marlinui.cpp]: I realize I have made a small mistake above, at the end of the three middle lines there is an extra `)` which should be deleted

File: Marlin/src/lcd/marlinui.cpp (hunk 2 above)
Comment (soligen2010) [reply Marlin/src/lcd/marlinui.cpp]: Good catch. I missed that too, and my config does not include those lines of code, so it compiled fine.

**Review by XDA-Bam — APPROVED** [review]

File: Marlin/src/lcd/marlinui.cpp [hunk 3]
```diff
@@ -1035,9 +1035,13 @@ void MarlinUI::init() {
...
-          TERN_(HAS_TOUCH_SLEEP, if (lastEncoderDiff != encoderDiff) wakeup_screen());
-          lastEncoderDiff = encoderDiff;
+          #if HAS_TOUCH_SLEEP
+            static int8_t lastEncoderDiff;
+            if (lastEncoderDiff != encoderDiff) {
+              wakeup_screen();
+              lastEncoderDiff = encoderDiff;
+            }
+          #endif
         #endif
```
Comment (classicrocker883) [inline Marlin/src/lcd/marlinui.cpp:1045]: actually would this change be better?

```diff
-        #if ENCODER_PULSES_PER_STEP > 1
-          #if HAS_TOUCH_SLEEP
             static int8_t lastEncoderDiff;
             if (lastEncoderDiff != encoderDiff) {
               wakeup_screen();
               lastEncoderDiff = encoderDiff;
             }
-          #endif
-        #endif

+        #if (ENCODER_PULSES_PER_STEP > 1) && HAS_TOUCH_SLEEP
           static int8_t lastEncoderDiff;
           if (lastEncoderDiff != encoderDiff) {
             wakeup_screen();
             lastEncoderDiff = encoderDiff;
           }
+        #endif
```

Comment (soligen2010) [reply Marlin/src/lcd/marlinui.cpp:1045]: Looks good to me.  I am not real well versed in GIT, but Looks like you already made these changes so I don't have to do anything.  LMK if you need something from me.

Comment (classicrocker883) [reply Marlin/src/lcd/marlinui.cpp:1045]: well I didn't actually make the changes, this is just a suggestion. you or a member has the authority to actually commit the change.

**Pull request merged**